procrastinate/admin.py

```diff
@@ -0,0 +1,174 @@
+from procrastinate import connector as connector_module
+from procrastinate import sql, utils
+
+
+@utils.add_sync_api
+class Admin:
+    """
+    The Admin is used to overview and administrate procrastinate jobs.
+
+    You should never need to instanciate an Admin object as it is
+    already available as `App.admin`.
+    """
+
+    def __init__(self, connector: connector_module.BaseConnector):
+        """
+        Parameters
+        ----------
+        connector:
+            Instance of a subclass of :py:class:`procrastinate.connector.BaseConnector`,
+            typically `PostgresConnector`. It will be responsible for all communications
+            with the database. Mandatory.
+        """
+        self.connector = connector
+
+    async def list_jobs_async(
+        self, id=None, queue=None, task=None, status=None, lock=None,
+    ):
+        """
+        List all procrastinate jobs given query filters.
+
+        Parameters
+        ----------
+        id: int
+            Filter by job ID
+        queue: str
+            Filter by job queue name
+        task: str
+            Filter by job task name
+        status: str
+            Filter by job status (*todo*/*doing*/*succeeded*/*failed*)
+        lock: str
+            Filter by job lock
+
+        Returns
+        -------
+        ``List[Dict[str, Any]]``
+            A list of dictionnaries representing jobs (``id``, ``queue``, ``task``,
+            ``lock``, ``args``, ``status``, ``scheduled_at``, ``attempts``).
+        """
+        return [
+            {
+                "id": row["id"],
+                "queue": row["queue_name"],
+                "task": row["task_name"],
+                "lock": row["lock"],
+                "args": row["args"],
+                "status": row["status"],
+                "scheduled_at": row["scheduled_at"],
+                "attempts": row["attempts"],
+            }
+            for row in await self.connector.execute_query_all(
+                query=sql.queries["list_jobs"],
+                id=id,
+                queue_name=queue,
+                task_name=task,
+                status=status,
+                lock=lock,
+            )
+        ]
+
+    async def list_queues_async(
+        self, queue=None, task=None, status=None, lock=None,
+    ):
+        """
+        List all queues and number of jobs per status for each queue.
+
+        Parameters
+        ----------
+        queue: str
+            Filter by job queue name
+        task: str
+            Filter by job task name
+        status: str
+            Filter by job status (*todo*/*doing*/*succeeded*/*failed*)
+        lock: str
+            Filter by job lock
+
+        Returns
+        -------
+        ``List[Dict[str, Any]]``
+            A list of dictionnaries representing queues stats (``name``, ``nb_jobs``,
```

Comment:
Minor but I think `num_jobs` makes for a better variable name. `nb_` is just in french, but I don't think it's common as an abbreviation for number in english. I might be wrong.

Reply:
Maybe it's just me but when I read `num` I think about "numero", so just a number without any context.
When I read `nb` I think about a counter. Then maybe `count_` would be more suitable?

Reply:
> Maybe it's just me but when I read num I think about "numero", so just a number without any context.

Yes, I know :-)

https://stackoverflow.com/questions/3742650/naming-conventions-for-number-of-foos-variables, for example, doesn't even mention `nb_` as a possible prefix. I really thing "nb_" is a french thing.

> Then maybe count_ would be more suitable?

Or as a suffix: `jobs_count`.

@ewjoachim any opinion about this?

Reply:
`jobs_count` is nice. It's not something we can't do on a second pass. If you want to change it, great, otherwise, I suggest we keep this in mind and make a follo-up PR. It's 1000 lines already !